corelib/main_api.py

```diff
@@ -113,6 +119,236 @@ def text_reco(image):
     return {"Text": preds}
 
 
+def greedyCaptionSearch(photo):
```

Comment:
Follow the same conventions as other places in the repo
Only the main function should be in main_api.py
every other utility function should be in a file inside corelib/<name_of_model>